### PR TITLE
[SG-397] Configure jest for Desktop and add simple tests

### PR DESCRIPTION
**Pull request by differsthecat**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Configure Jest for Desktop and add simple tests for an example

### Configuration
- **apps/desktop/jest.config.js**
- **apps/desktop/package.json** 
- **apps/desktop/test.setup.ts**
- ~~**apps/desktop/tsconfig.json:** Had to add Jest types here or it would complain in the test files~~ Environment issue, removed.
- **apps/desktop/tsconfig.spec.json**
- **jest.config.js** Added desktop :D 

### Tests
- **apps/desktop/src/app/vault/generator.component.spec.ts:** Add simple tests using jest-mock-extended to demonstrate dependency injection and testing a service call

## Screenshots
Yay green!
Screen Shot 2022-09-08 at 10 27 47 AM

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/desktop/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "module": "es6",
     "target": "ES2016",
     "sourceMap": true,
-    "types": [],
+    "types": ["jest"],
```
Comment (Hinton) [inline apps/desktop/tsconfig.json:10]: I think you can just remove this line completely and it should automatically find the jest typings.

Comment (differsthecat) [reply apps/desktop/tsconfig.json]: It wasn't working for some reason for me locally unless I added this; I just quit VSCode and reopened it and it no longer complains. Removed 👍

**Review by gbubemismith — APPROVED** [review]
this looks good. Great job!

**Comment by differsthecat** [comment]
Thank you @gbubemismith !

**Pull request merged**